api_tests/test_functional.py

```diff
@@ -32,3 +39,51 @@ def test_support_selective_serialization():
     Only serializable globals are returned; others are simply omitted.
     """
     assert {"out2": "hi"} == call_api_success("out1 = object(); out2 = 'hi'", {})
+
+
+def test_support_exception_traceback_stderr():
+    """
+    Exceptions from submitted code are serialized and returned, along with stderr.
+
+    Globals aren't updated when there's an exception, even via mutation.
+    """
+    (globals_out, emsg) = call_api_code_error(dedent("""
+      # Try updating some globals (won't work)
+      primitive = 22
+      container.append(123)
+
+      # stdout is suppressed by default (due to implementation detail -- codejail
+      # delivers globals updates on stdout as JSON)
+      print("some output to stdout --- suppressed")
+
+      # stderr will be returned, though
+      import sys
+      print("other output to stderr", file=sys.stderr)
+
+      # Raise an exception
+      1/0
+    """), {"primitive": 7, "container": []})
+
+    # The codejail library isn't actually able to provide updated globals when
```

Comment:
Maybe some of these comments should be moved or duplicated in the docstring about the test?

Reply:
I have a preference to keep them closer to the assertion in question. It's really an unspecified aspect of codejail and I'd like to keep it as much of a side-note as possible.

Reply:
You can leave these, but I added another comment for the docstring.